api/v1beta1/policyautomation_types.go

```diff
@@ -14,14 +14,15 @@ type PolicyAutomationSpec struct {
 	// +kubebuilder:validation:Required
 	PolicyRef string `json:"policyRef"`
 	// Mode decides how automation is going to be triggered
-	// +kubebuilder:validation:Enum={once,disabled}
+	// +kubebuilder:validation:Enum={once,everyEvent,disabled}
 	// +kubebuilder:validation:Required
 	Mode string `json:"mode"`
```

Comment:
One of the things I was hoping to do as part of this feature is to make `Mode` a custom type rather than `string`. You can call the type `PolicyAutomationMode` and then define constants for each value in the enum.

This is the standard approach for defining Kubernetes APIs but was not done in the original implementation here.